inc/pdu.class.php

```diff
@@ -0,0 +1,254 @@
+      echo "<td>";
+      Location::dropdown([
+         'value'  => $this->fields["locations_id"],
+         'entity' => $this->fields["entities_id"],
+         'rand'   => $rand
+      ]);
+      echo "</td>";
+      echo "<td><label for='dropdown_pdutypes_id$rand'>".__('Type')."</label></td>";
+      echo "<td>";
+      PDUType::dropdown([
+         'value'  => $this->fields["pdutypes_id"],
+         'rand'   => $rand
+      ]);
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td><label for='dropdown_users_id_tech$rand'>".__('Technician in charge of the hardware')."</label></td>";
+      echo "<td>";
+      User::dropdown([
+         'name'   => 'users_id_tech',
+         'value'  => $this->fields["users_id_tech"],
+         'right'  => 'own_ticket',
+         'entity' => $this->fields["entities_id"],
+         'rand'   => $rand
+      ]);
+      echo "</td>";
+      echo "<td><label for='dropdown_manufacturers_id$rand'>".__('Manufacturer')."</label></td>";
+      echo "<td>";
+      Manufacturer::dropdown([
+         'value' => $this->fields["manufacturers_id"],
+         'rand' => $rand
+      ]);
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td><label for='dropdown_groups_id_tech$rand'>".__('Group in charge of the hardware')."</label></td>";
+      echo "<td>";
+      Group::dropdown([
+         'name'      => 'groups_id_tech',
+         'value'     => $this->fields['groups_id_tech'],
+         'entity'    => $this->fields['entities_id'],
+         'condition' => '`is_assign`',
+         'rand'      => $rand
+      ]);
+
+      echo "</td>";
+      echo "<td><label for='dropdown_pdumodels_id$rand'>".__('Model')."</label></td>";
+      echo "<td>";
+      PDUModel::dropdown([
+         'value'  => $this->fields["pdumodels_id"],
+         'rand'   => $rand
+      ]);
+      echo "</td></tr>\n";
+
+      echo "<td><label for='textfield_serial$rand'>".__('Serial number')."</label></td>";
+      echo "<td >";
+      Html::autocompletionTextField($this, 'serial', ['rand' => $rand]);
+      echo "</td>";
+
+      echo "<td><label for='textfield_otherserial$rand'>".sprintf(__('%1$s%2$s'), __('Inventory number'), $tplmark).
+           "</label></td>";
+      echo "<td>";
+
+      $objectName = autoName($this->fields["otherserial"], "otherserial",
+                             (isset($options['withtemplate']) && ($options['withtemplate'] == 2)),
+                             $this->getType(), $this->fields["entities_id"]);
+      Html::autocompletionTextField(
+         $this,
+         'otherserial',
+         [
+            'value'     => $objectName,
+            'rand'      => $rand
+         ]
+      );
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td><label for='comment'>".__('Comments')."</label></td>";
+      echo "<td colspan='3' class='middle'>";
+
+      echo "<textarea cols='45' rows='3' id='comment' name='comment' >".
```

Comment:
not mandatory, but we have a function ``Html::textarea``

Reply:
Just quickly tested... Seems that there are issues with labels (name is not part of the id, rand seems ignored, ...) :(